src/components/TokenEnterAmount.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useMemo, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import {
+  Platform,
+  TextInput as RNTextInput,
+  StyleProp,
+  StyleSheet,
+  Text,
+  TextStyle,
+  View,
+} from 'react-native'
+import { getNumberFormatSettings } from 'react-native-localize'
+import TextInput from 'src/components/TextInput'
+import TokenDisplay from 'src/components/TokenDisplay'
+import TokenIcon, { IconSize } from 'src/components/TokenIcon'
+import Touchable from 'src/components/Touchable'
+import DownArrowIcon from 'src/icons/DownArrowIcon'
+import SwapArrows from 'src/icons/SwapArrows'
+import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { getLocalCurrencySymbol } from 'src/localCurrency/selectors'
+import { useSelector } from 'src/redux/hooks'
+import { type AmountEnteredIn } from 'src/send/types'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { type TokenBalance } from 'src/tokens/slice'
+
+export const APPROX_SYMBOL = '≈'
+
+const BORDER_RADIUS = 12
+
+function groupNumber(value: string) {
+  return value.replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, 'group')
+}
+
+export default function TokenEnterAmount({
+  token,
+  inputValue,
+  tokenAmount,
+  localAmount,
+  amountType,
+  inputRef,
+  inputStyle,
+  autoFocus,
+  editable = true,
+  testID,
+  onInputChange,
+  toggleAmountType,
+  onOpenTokenPicker,
+}: {
+  token?: TokenBalance
+  inputValue: string
+  tokenAmount: string
+  localAmount: string
+  amountType: AmountEnteredIn
+  inputRef: React.MutableRefObject<RNTextInput | null>
+  inputStyle?: StyleProp<TextStyle>
+  autoFocus?: boolean
+  editable?: boolean
+  testID?: string
+  onInputChange(value: string): void
+  toggleAmountType?(): void
+  onOpenTokenPicker?(): void
+}) {
+  const { t } = useTranslation()
+  // the startPosition and inputRef variables exist to ensure TextInput
+  // displays the start of the value for long values on Android
+  // https://github.com/facebook/react-native/issues/14845
+  const [startPosition, setStartPosition] = useState<number | undefined>(0)
+  // this should never be null, just adding a default to make TS happy
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  const tokenPlaceholder = new BigNumber(0).toFormat(2)
+  const localPlaceholder = `${localCurrencySymbol}${new BigNumber(0).toFormat(2).replaceAll('.', decimalSeparator)}`
+
+  const formattedInputValue = useMemo(() => {
+    const number = groupNumber(inputValue)
+      .replaceAll('.', decimalSeparator)
+      .replaceAll('group', groupingSeparator)
+    if (amountType === 'token') return number
+    return number !== '' ? `${localCurrencySymbol}${number}` : ''
+  }, [inputValue, amountType, localCurrencySymbol])
```

Comment:
I figured its easier to add currency sign before whatever is typed in the amount field rather than trying to parse it while having that sign and manage all the use-cases when it whether includes the sign (when typing in fiat) or not (when typing in crypto).